tfs/src/main/java/hudson/plugins/tfs/TeamCollectResultsPostBuildAction.java

```diff
@@ -82,7 +80,7 @@ public void perform(
             final String folderName = teamResultType.getFolderName();
             logger.print(" " + teamResultType.getDisplayName());
             final File resultFolder = new File(resultsRoot, folderName);
-            resultFolder.mkdirs();
+            Files.createDirectory(resultFolder.toPath());

```

Comment:
To be consistent with the previous behavior, I think you should use Files.createDirectories (since you are replacing mkdirs). This will create any parent directories that don't exist.